client/my-sites/plans/plans.jsx

```diff
@@ -0,0 +1,28 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+
+/**
+ * Internal Dependencies
+ */
+import config from 'config';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import getIntervalTypeForTerm from 'state/plans/get-interval-type-for-term';
+import getCurrentPlanTerm from 'state/selectors/get-current-plan-term';
+import Plans from 'my-sites/plans/main';
+
+export default connect( ( state, { intervalType } ) => {
+	// For WP.com plans page, if intervalType is not explicitly specified in the URL,
+	// we want to show plans of the same term as plan that is currently active
+	if ( ! intervalType && config.isEnabled( 'upgrades/2-year-plans' ) ) {
```

Comment:
As noted in the previous comments, we could move this to a separate utility function with its own unit tests.